Rails support (and better Sinatra support)

samstokes opened the pull request:
Improves this middleware's functionality when used with a Rails app, capturing the Rails controller, action and matched route pattern (in fields `request.controller`, `request.action` and `request.route`) and request params (in `request.params.*`) as well as the literal URL requested.

A few other improvements thrown in:

 * likewise captures `request.route` for Sinatra apps too (but not the request params, those aren't so easy to get at from a mere middleware)
 * fixes a bug where some fields weren't getting set if the request threw an exception
 * CI build tests on Rails versions 3 through 5

Commit:
Clean up logging and gem detection

Commit:
Record route (as well as concrete path) for Sinatra apps

Commit:
Remove unused ivar

Commit:
For Sinatra apps, set 'name' to the route

Commit:
For Rails apps, log controller and action

Commit:
Tests for Sinatra route logging

Commit:
Report named route for Rails too (at some effort)

Commit:
Tests for Rails support

Commit:
Test against multiple Rails versions

Commit:
Make specs pass on Rails 3

Commit:
report more specific meta.package where possible

Commit:
Include more fields for erroring requests

martin308 committed:
Fix warn logger method

martin308 commented on lib/rack-honeycomb/auto_install.rb:
```diff
@@ -56,8 +67,35 @@ def add_honeycomb_field(field, value)
+          @logger.debug "#{self.name}: #{msg}" if @logger
+        end
+
+        def warn(*args)
```
Looks like this argument might cause an issue, will push up a fix

martin308 committed:
Require shared examples in spec_helper

Previously you couldn't run some of the specs individually as they
depended on the the plain_rack_spec including these shared examples
first

martin308 committed:
Fix code sample

martin308 committed:
Include sinatra flag

martin308 committed:
Include rails flag

martin308 committed:
Encourage usage of beeline-ruby

martin308 reviewed (approved):
LGTM
